snorkel/slicing/monitor.py

```diff
@@ -0,0 +1,120 @@
+from typing import Dict, List, Union
+
+import numpy as np
+import pandas as pd
+
+from snorkel.classification.scorer import Scorer
+from snorkel.labeling.lf import LabelingFunction
+from snorkel.slicing import PandasSFApplier
+
+SliceMetricsDict = Dict[str, Dict[str, float]]
+
+
+class PandasSlicer:
+    """Create DataFrames corresponding to slices.
+
+    Parameters
+    ----------
+    df
+        A pandas DataFrame that will be sliced
+
+    Attributes
+    ----------
+    df
+        See above.
+    """
+
+    def __init__(self, df: pd.DataFrame):
+        self.df = df
+
+    def slice(self, slicing_function: LabelingFunction) -> pd.DataFrame:
+        """Return a dataframe with examples corresponding to specified slice_name.
+
+        Parameters
+        ----------
+        slicing_function
+            slicing_function which will operate over self.df to return a subset of examples
+
+        Returns
+        -------
+        pd.DataFrame
+            A DataFrame including only examples belonging to slice_name
+        """
+
+        S_matrix = PandasSFApplier([slicing_function]).apply(self.df)
+
+        # Index into the SF labels for the first (and only) column
+        df_idx = np.where(S_matrix[:, 0])[0]
+        return self.df.iloc[df_idx]
+
+
+class SliceScorer:
+    """Scorer that returns metrics on overall performance and slices.
```

Comment:
"...as well as slices."

Reply:
Just to emphasize that that's the difference here from a normal scorer.

Reply:
👍